torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1475,6 +1488,43 @@ def solve(self, disable_congruences=True, disable_equivalences=True):
         for source, expr in self._symbolic_equivalences:
             self._dynamic_results.add(f"{self._dcp.print_source(source)} == {self._dcp.doprint(expr)}")
 
+    @classmethod
+    def is_supported_equivalence(cls, expr):
+        # Currently supported Dim ops are linear expressions with integer coefficients.
+        # So check that expr only contains +, *, ints, and a single occurrence of a symbol
+        if isinstance(expr, (sympy.Add, sympy.Mul)):
+            lhs, rhs = expr.args
+            return (
+                (cls.is_supported_equivalence(lhs) and isinstance(rhs, sympy.Integer)) or
+                (isinstance(lhs, sympy.Integer) and cls.is_supported_equivalence(rhs))
+            )
+        return isinstance(expr, sympy.Symbol)
+
+    @classmethod
+    def solve_supported_equivalence(cls, expr, solution):
```

Comment:
I'm not entirely clear what the signature of this function is, are we solving `expr == solution` or something else?

Reply:
Using try_solve now.